update markdig-engine from master

cillroy opened the pull request:
## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

mairaw committed:
fix broken links

mairaw committed:
fix formatting

mairaw committed:
fixed *

mairaw committed:
markdig conversion - remove table with code

BillWagner committed:
markdig fixes in winforms (#8508)

Two different list indicators with no content.

BillWagner committed:
Markdig fixes workflow (#8510)

* code fence the email address so it's not parsed as an HTML link

* remove extra parentheses.

* new lines for list and note

One note was not displayed correctly.

The list above it is meant to be separated by new lines, but was not.

* add blank line so Note formats correctly

BillWagner committed:
Markdig fixes wcf (#8512)

* Fix snippet that should have been a config file

This is a bit of a guess.

* table cell separators not surrounded by spaces

Markdig fixes.

* warnings on subsequent code blocks.

* code fence ~ character

BillWagner committed:
More WPF markdig fixes

Commit:
Markdig migration fixes

mairaw committed:
Merge pull request #8678 from dotnet/port-markdig-changes

Merge markdig changes into master

KristianJakubik committed:
Add missing Async suffix to ProcessURL method (#8637)

* Add missing Async suffix to ProcessURL method

* Rename additional ProcessURL to ProcessURLAsync

giotavekiari committed:
MyGet Plan&Link Change (#8615)

* MyGet Plan Change

Issue #8386

* Update publish-nuget-package.md

Thraka committed:
dotnet test describes RunSettings (#8618)

* dotnet test describes RunSettings

* updated description

imba-tjd committed:
Else -> else (#8723)

eonarheim committed:
Update `Search()` formatting to be consistent with earlier usage (#8624)

* Update formatting to be consistent with earlier usage

* Update to `Search` per feedback

douglaslMS committed:
Cleaned up topic, fixed inline code samples esp USINGs. (#8651)

* Cleaned up topic, fixed inline code samples esp USINGs.

* Made almost all changes suggested by reviewer.

TheSench committed:
Link integral types to relevant language spec page. (#8693)

TheSench committed:
Link various operators to relevant language spec page. (#8696)

TheSench committed:
Link access modifiers to relevant language spec page. (#8697)

TheSench committed:
Link various pages about classes to relevant language spec page. (#8698)

Drookoo committed:
removed the redundant link in issue #8703  (#8713)

* Removed redundant links #8703

* Removed redundant links #8703 and tmp file

TheSench committed:
Issue 6886 programming guides (#8699)

* Link various pages about classes to relevant language spec page.

* Link various programming guides to relevant language spec page.

nemrism committed:
removing old Technet links (#8717)

daveabrock committed:
typo fix (#8732)

cartermp committed:
Add intro to functional programming (#8367)

* Add intro to FP

* Add to TOC and modify date

* Remove the why fp section

* Acrolinx suggestions

* Update link

* Rename first-class functions article, redirect, and link to it with button

* Feedback and more comparisons

* Remove needless statement

* Fully rename the first class functions article

* Feedback and update dates

* Fix build

* Swap redirection

* Update ternary bit

* remove dumb div

user135711 committed:
typo (#8735)

zackasattackas committed:
Add link to boxing guide (#6351) (#8737)

* Add link to boxing guide (#6351)

* Remove locale from hyperlink

Commit:
Fixed broken links (#8738)

lucasnobredev committed:
whitespace new cases in connection-strings (#8741)

jmarkman committed:
LINQ to XML Object Loading step (#8715)

* added object loading step
- added method syntax versions of queries
- added path construction step and loading step
- added link to How to: Load XML from a File (C#)

* moved link to more appropriate section and added stream link

* updated based on feedback
- implemented platform-agnostic file loading code snippet
- added cast on the Select method at line 38
- updated ms.date

* i guess i didn't include the cast in the previous commit

dibbdob committed:
Update dotnet-publish.md (#8318)

* Update dotnet-publish.md

* Added clarification

MartinWilkerson committed:
Mark IClassFixture as code in prose (#8679)

* Mark IClassFixture as code in prose

`IClassFixture<WebApplicationFactory<TEntry>>` was rendering as `IClassFixture<WebApplicationFactory>`. Adding backquotes gets it to render the missing `<TEntry>` part at least.

* Escape angle brackets instead of fencing

Escaping the angle brackets makes it consistent with the rest of the document.

pkulikov committed:
Revise the AND operator articles (#8687)

* More structure to the add operator articles

* Revised AND operator topic

* Revised AND assignment operator

* Addressed feedback

karelz committed:
UDP service - fix sample descriptions (#8729)

* UDP service - fix sample descriptions

Fix sample descriptions

Additional changes:
* Remove extra spaces at the start and end of text and code.
* Simplify code samples

* Code review feedback

jgscherber committed:
Format issue (missing code formatting around ==) (#8753)

gsomix committed:
Update F# optional parameters docs (#8746)

* Update F# optional parameters docs

* Update parameters-and-arguments.md

* Update parameters-and-arguments.md

* Update parameters-and-arguments.md

* Update parameters-and-arguments.md

* Update parameters-and-arguments.md

zochris committed:
fix little typo in Auto-Implemented Properties (#8754)

Commit:
Update dispatch-by-body-element.md (#8747)

* Update dispatch-by-body-element.md

the closing brace in the code block is added

* Added language identifiers.

focuse committed:
Fix typo in generic-interfaces.md (#8755)

* Fix typo in generic-interfaces.md

fixed a simple typo, that is **"contravariant"** is a single word.

* Fixed typo in generic-interfaces.md

jrstrayhorn committed:
Update broken links (#8660)

* Update broken links

Updating link to ListView with Multiple Views Sample

* Removing localization

Removing localization en-us

* making link changes based on recommendations

changing link based on recommendation

* removing broken links

removing broken links based on recommendation

nemrism committed:
Replace some MSDN links (#8718)

* replace some MSDN links

* address commit validation error

* Update index.md

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=8786) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=8786) before we can accept your contribution.<br/><br/>:x: cillroy [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=8786)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=8786) it.</sub>

lucasnobredev committed:
Change absolute https://docs.microsoft.com/dotnet links to relative links (#8740)

* relative links

* correct choosing-between-class-and-struct

* improvements from pull request

* remove md

randomnote1 committed:
Added missing backslash to PowerShell command (#8760)

Added a missing backslash to the PowerShell command in the "To check for a minimum-required .NET Framework version by querying the registry in PowerShell (.NET Framework 4.5 and later)" section.

code4t2 committed:
Updated URL for extension methods to internal documentation rather than (#8783)

the codeproject repo

code4t2 committed:
Fixed URl of 'nameof' operator (#8784)

* Updated URL for extension methods to internal documentation rather than
the codeproject repo

* Updated URL for 'nameof' operator to internal documentation rather than
the stackoverflow page

MoxieHampton committed:
Made minor style edit (#8787)

Ensured caveats were uniformed.

gewarren committed:
Add info about new checkbox in Visual Studio (#8763)

* Add info about new checkbox in Visual Studio

* Boldface .config file names

Saisang committed:
markdig manual fix render issue (#8791)

Commit:
Fix typo describing TripleDES as an asymmetric algorithm (#8795)

jiayi11 committed:
Add new help page for WCF trace: warnservicehealthenablednobaseaddress (#8762)

* add new help page for wcf trace: warnservicehealthenablednobaseaddress

* remove assetid line

* update toc and trace-reference files

pkulikov committed:
Revised the assignment operator article (#8790)

huanwu committed:
Add dotnet-svcutil.xmlserializer guide (#8734)

* Add dotnet-svcutil.xmlserializer guide

* Fix based on the comments

ejball committed:
Consider keeping obsolete API (#8683)

rpetrusha reviewed (approved):
I'll merge your PR now, @cillroy
